staging/src/k8s.io/kubectl/pkg/cmd/config/delete_user.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// DeleteUserOptions holds the data needed to run the command
+type DeleteUserOptions struct {
+	user string
+
+	configAccess clientcmd.ConfigAccess
+	config       *clientcmdapi.Config
+	configFile   string
+
+	genericclioptions.IOStreams
+}
+
+var (
+	deleteUserExample = templates.Examples(`
+		# Delete the minikube user
+		kubectl config delete-user minikube`)
+)
+
+// NewDeleteUserOptions creates the options for the command
+func NewDeleteUserOptions(ioStreams genericclioptions.IOStreams, configAccess clientcmd.ConfigAccess) *DeleteUserOptions {
+	return &DeleteUserOptions{
+		configAccess: configAccess,
+		IOStreams:    ioStreams,
+	}
+}
+
+// NewCmdConfigDeleteUser returns a Command instance for 'config delete-user' sub command
+func NewCmdConfigDeleteUser(streams genericclioptions.IOStreams, configAccess clientcmd.ConfigAccess) *cobra.Command {
+	o := NewDeleteUserOptions(streams, configAccess)
+
+	cmd := &cobra.Command{
+		Use:                   "delete-user NAME",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Delete the specified user from the kubeconfig"),
+		Long:                  "Delete the specified user from the kubeconfig",
+		Example:               deleteUserExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	return cmd
+}
+
+// Complete sets up the command to run
+func (o *DeleteUserOptions) Complete(cmd *cobra.Command, args []string) error {
+	config, err := o.configAccess.GetStartingConfig()
+	if err != nil {
+		return err
+	}
+	o.config = config
+
+	if len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, "user to delete is required")
+	}
+	o.user = args[0]
+
+	configFile := o.configAccess.GetDefaultFilename()
+	if o.configAccess.IsExplicitFile() {
+		configFile = o.configAccess.GetExplicitFile()
+	}
+	o.configFile = configFile
+
+	return nil
+}
+
+// Validate ensures the command has enough info to run
+func (o *DeleteUserOptions) Validate() error {
+	_, ok := o.config.AuthInfos[o.user]
```

Comment:
can o.config be nil?  If I never called Complete() it seems like it could be.

Reply:
Paired with @soltysh and we agreed that it would be a programmer error if this was `nil` and not a user error.

The expectation is a programmer calling this elsewhere should have set the object up correctly and it's easy to catch.